### ISSUE TITLE
Add option for notifications to apply only on specified branches

### ISSUE THREAD

**Issue by OJFord** [travis-ci/travis-ci#6084]
It would be great to be able to do:

``` yaml
notifications:
  slack:
    branches:
      - 'master'
      - 'foobar'
```

If left unspecified of course defaulting to the current 'all' behaviour.

**Comment by OJFord** [comment]
Actually, there's some previous discussion at #1405, which was subsequently closed without comment.

Does that mean this idea has been considered and dismissed? @roidrage commented in that thread sounding keen to at least support something like `only_master: true`.

**Comment by BanzaiMan** [comment]
We have an internal ticket to track this feature request, and will share updates in #1405 when we have any.

Thanks.

**Comment by bcruddy** [comment]
@BanzaiMan is there still a feature request for this? @DrTorte closed #1405 three weeks ago without a comment.

**Comment by DrTorte** [comment]
Hey there, no updates on this yet. It's still something we'd like to do so it remains open internally.
